jwtkms/common.go

```diff
@@ -1,20 +1,44 @@
 package jwtkms
 
-import "github.com/aws/aws-sdk-go/service/kms"
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+)
 
 const messageTypeDigest = "DIGEST"
 
-// KmsConfig is a struct to be passed to token signing/verification
-type KmsConfig struct {
-	//A configured svc pointer to AWS KMS
-	Svc *kms.KMS
+// Config is a struct to be passed to token signing/verification.
+type Config struct {
+	// context used for kms operations
+	ctx context.Context
+
+	// A configured kms client pointer to AWS KMS
+	kmsClient *kms.Client
 
-	//AWS KMS Key ID to be used
-	KmsKeyId string
+	// AWS KMS Key ID to be used
+	kmsKeyID string
 
-	//If set to true JWT verification will be performed using KMS's Verify method
+	// If set to true JWT verification will be performed using KMS's Verify method
 	//
-	//In normal scenarios this can be left on the default false value, which will get, cache(forever) in memory and
-	//use the KMS key's public key to verify signatures
-	VerifyWithKMS bool
+	// In normal scenarios this can be left on the default false value, which will get, cache(forever) in memory and
+	// use the KMS key's public key to verify signatures
+	verifyWithKMS bool
+}
+
+// NewKMSConfig create a new Config with specified parameters.
+func NewKMSConfig(client *kms.Client, keyID string, verify bool) *Config {
+	return &Config{
+		ctx:           context.Background(),
+		kmsClient:     client,
+		kmsKeyID:      keyID,
+		verifyWithKMS: verify,
+	}
+}
+
+// WithContext adds a context to Config.
+func (c *Config) WithContext(ctx context.Context) *Config {
```

Comment:
I think making a copy of the `Config` would be less error-prone (otherwise we're actually modifying the same `Config`, which might lead to non-obvious context issues)
```suggestion
func (c Config) WithContext(ctx context.Context) Config {
```